### ISSUE TITLE
Multiple partitions don't mount 🐛 [Addon name] <title>

### ISSUE THREAD

**Issue by open-forest** [dianlight/hassio-addons#185]
### Addon

SambaNAS

### Description

If a USB disk contains multiple labeled partitions, the partitions are found, but they do not get mounted.
In my case I have three partitions: backup (a reserved name, so I will have to change the label), frigate, media. As you can see in the log, all three labels are found, the reserved name is filtered out, and then the names frigate and media are combined and an attempt is made to find the disk labeled "frigate media", which fails.

### Reproduction steps

```bash
1. Go to 'Configuration'
2. Click on 'Automount'
3. Save & Restart
4. See error
```

### Addon Logs

```bash
---------------------------------------------------
[19:55:51] INFO: Available Disk Labels:
[19:55:51] INFO: 	backup[ext4]
[19:55:51] INFO: 	media[ext4]
[19:55:51] INFO: 	frigate[ext4]
---------------------------------------------------
[19:55:51] INFO: MoreDisks to mount:

	frigate
[19:55:51] INFO: Automount is Enabled!
---------------------------------------------------
[19:55:51] INFO: Checking Mounting disks for reserved names:

	backup
	frigate
	media
---------------------------------------------------
---------------------------------------------------
[19:55:51] INFO: Mounting disks:

	 frigate media
---------------------------------------------------
[19:55:51] INFO: Disk  frigate media not found! <SKIP>
s6-rc: info: service init-automount successfully started
s6-rc: info: service init-smartd: starting
s6-rc: info: service init-samba: starting
s6-rc: info: service hd-idle: starting
s6-rc: info: service hd-idle successfully started
[19:55:51] INFO: Enabling S.M.A.R.T for /dev/sda
smartctl 7.3 2022-02-28 r5338 [x86_64-linux-6.1.63-haos] (local build)
Copyright (C) 2002-22, Bruce Allen, Christian Franke, www.smartmontools.org

[19:55:51] INFO: Hostname: homeassistant
=== START OF ENABLE/DISABLE COMMANDS SECTION ===
SMART Enabled.
SMART Attribute Autosave Enabled.
SMART Automatic Offline Testing Enabled every four hours.

[19:55:51] INFO: Enabling S.M.A.R.T for /dev/sdb
smartctl 7.3 2022-02-28 r5338 [x86_64-linux-6.1.63-haos] (local build)
Copyright (C) 2002-22, Bruce Allen, Christian Franke, www.smartmontools.org

[19:55:51] INFO: Interfaces: wlp1s0 enp2s0 
=== START OF ENABLE/DISABLE COMMANDS SECTION ===
SMART Enabled.
SMART Attribute Autosave Enabled.
SMART Automatic Timers not supported
```

### Addon Config

```bash
workgroup: WORKGROUP
username: homeassistantnas
allow_hosts:
  - 10.0.0.0/8
  - 172.16.0.0/12
  - 192.168.0.0/16
  - 169.254.0.0/16
  - fe80::/10
  - fc00::/7
automount: true
moredisks:
  - frigate
mountoptions:
  - nosuid
  - relatime
  - noexec
available_disks_log: true
medialibrary:
  enable: false
veto_files:
  - ._*
  - .DS_Store
  - Thumbs.db
  - icon?
  - .Trashes
compatibility_mode: false
recyle_bin_enabled: false
wsdd2: false
mqtt_use_legacy_entities: false
autodiscovery: {}
other_users: []
acl: []
interfaces: []
password: xxx
```

### Architecture

other

### OS

HaOS

**Comment by LaStrada** [comment]
Same issue here. I have:
* lastrada-backup
* tina-backup

And I get this:
`[14:03:57] INFO: Disk lastrada- tina-backup not found! <SKIP>`
For some reason the `backup` is only removed from the first partition name. This was not the case before, both shares were mounting just fine.

I can easily just rename the partitions if that's a problem, but I still have the same issue with no partitions mounted, and it seems like it's combined into one name.

**Comment by dianlight** [comment]
Fixed in 12.2.0-nas #191
